src/main/java/reposense/model/CliArguments.java

```diff
@@ -508,7 +461,9 @@ public Builder reportConfiguration(ReportConfiguration reportConfiguration) {
          * @return CliArguments
          */
         public CliArguments build() {
-            return new CliArguments(this);
+            CliArguments built = this.cliArguments;
+            this.cliArguments = new CliArguments();
```

Comment:
Can I clarify why we choose to reset `cliArguments` here?

Reply:
Sure! The main motivation behind resetting the Builder object on every `build` operation is to prevent aliasing issues when the user chooses to rebuild `CliArguments` multiple times. Since the cloning operation has not been implemented yet, we decided that the temporary fix would be to reset the Builder object on every build.

A new issue has been released which aims to tackle this issue and to allow duplication of `CliArguments` to avoid resetting the Builder object on every build.